### PR TITLE
Updated null stats tests to include data initialized in setUpClass

### PR DESCRIPTION
**Pull request by jacob-buehler**
Updated null stats tests to include data initialized in setUpClass function

### CODE REVIEW

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 1]
```diff
@@ -33,8 +33,7 @@
     StructuredOptions,
     UnstructuredOptions,
 )
-
-from . import utils as test_utils
+from dataprofiler.tests.profilers import utils as test_utils
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:36]: why did this need to change?

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 2]
```diff
@@ -3592,8 +3602,10 @@ def test_correct_rows_ingested(self):
         )
 
     def test_correct_null_row_counts(self):
-        file_path = os.path.join(test_root_path, "data", "csv/empty_rows.txt")
-        data = pd.read_csv(file_path)
+        # file_path = os.path.join(test_root_path, "data", "csv/empty_rows.txt")
+        # data = pd.read_csv(file_path)
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:3606]: is this needed?

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 3]
```diff
@@ -3602,18 +3614,25 @@ def test_correct_null_row_counts(self):
...
+
+        # file_path = os.path.join(test_root_path, "data", "csv/iris-with-null-rows.csv")
+        # data = pd.read_csv(file_path)
+        # data = self.data
+
+        # profile = dp.StructuredProfiler(data, options=profiler_options)
+        # self.assertEqual(13, profile.row_has_null_count)
+        # self.assertEqual(13 / 24, profile._get_row_has_null_ratio())
+        # self.assertEqual(3, profile.row_is_null_count)
+        # self.assertEqual(3 / 24, profile._get_row_is_null_ratio())
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:3635]: i'd recommend either deleting or keeping -- not just keeping commented out code in the file

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 4]
```diff
@@ -3646,35 +3665,39 @@ def test_null_in_file(self):
                 "row_statistics.is_enabled": True,
             }
         )
-        data = dp.Data(filename_null_in_file)
+        # data = dp.Data(filename_null_in_file)
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:3668]: do we need to keep commented out?

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 5]
```diff
@@ -3646,35 +3665,39 @@ def test_null_in_file(self):
...
+        data = self.data
+
         profile = dp.StructuredProfiler(data, options=profiler_options)
 
         report = profile.report(report_options={"output_format": "pretty"})
-        count_idx = report["global_stats"]["profile_schema"]["COUNT"][0]
-        numbers_idx = report["global_stats"]["profile_schema"][" NUMBERS"][0]
+        names_idx = report["global_stats"]["profile_schema"]["names"][0]
+        numbers_idx = report["global_stats"]["profile_schema"]["numbers"][0]
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:3675]: changing becuase of new dataset

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 6]
```diff
@@ -3646,35 +3665,39 @@ def test_null_in_file(self):
...
+        #     ["test1", 1.0],
+        #     ["test2", 2.0],
+        #     ["test3", 3.0],
+        #     [None, None],
+        #     ["test5", 5.0],
+        #     ["test6", 6.0],
+        #     [None, None],
+        #     ["test7", 7.0],
+        # ]
+        # data = pd.DataFrame(data, columns=["NAME", "VALUE"])
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:3698]: same as above on commented out code

File: dataprofiler/tests/profilers/test_profile_builder.py (hunk 1 above)
Comment (jacob-buehler) [reply dataprofiler/tests/profilers/test_profile_builder.py:36]: the original line didn't import utils properly on my local, but the new line imports it consistently

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 7]
```diff
@@ -3771,28 +3803,36 @@ def test_null_row_stats_correct_after_updates(self, *mocks):
...
+
+        self.assertEqual(2, profile.row_has_null_count)
+        self.assertEqual(0, profile.row_is_null_count)
+        self.assertEqual(1.0, profile._get_row_has_null_ratio())
+        self.assertEqual(0, profile._get_row_is_null_ratio())
+        self.assertEqual(2, profile._min_sampled_from_batch)
+        # self.assertSetEqual({}, profile._profile[0].null_types_index)
+        # self.assertSetEqual({}, profile._profile[1].null_types_index)
+        self.assertEqual({}, profile._profile[0].null_types_index)
+        self.assertEqual({}, profile._profile[1].null_types_index)
```
[the remark this answers is not in the document]
Comment (jacob-buehler) [reply dataprofiler/tests/profilers/test_profile_builder.py:3817]: to solve an issue: 'dict' has no attribute 'difference'

File: dataprofiler/tests/profilers/test_profile_builder.py (hunk 1 above)
Comment (JGSweets) [reply dataprofiler/tests/profilers/test_profile_builder.py:36]: That might be bc the test was run from the dir as opposed to base of repo. I think here we are trying to separate from test vs in lib by using relative. but open to convo if we need a change here.

**Review by JGSweets — CHANGES_REQUESTED** [review]
I think we may need to take a step back here and identify the need of this PR. 
Identify which tests in here are slow and work towards what we need to do to speed those up as oppose to a changing everything. I'm concerned by unifying to a single data source we are losing data variability which has intention within the structure of the data to test different cases of profiling.

**Comment by taylorfturner** [comment]
Related to issue #866

**Pull request closed without merging**